fw/cache.c

```diff
@@ -1332,10 +1365,12 @@ tfw_cache_h2_copy_hdr(TfwCacheEntry *ce, char **p, TdbVRec **trec, TfwStr *hdr,
 				return -ENOMEM;
 		}
 
-		if (tfw_cache_h2_copy_int(&ce->hdr_len, s_val.len, 0x7f, p,
+		s_val_len = (need_extra_quotes ? s_val.len + 2 : s_val.len);
```

Comment:
```suggestion
		s_val_len =  s_val.len + 2 * need_extra_quotes;
```
In my understanding since `bool` isn't a native C type, GCC still can do the optimization and generates a conditional jump